### PR TITLE
Fix up approx 20% of the rubocop issues in aruba

### PR DESCRIPTION
**Pull request by luke-hill**
## Summary

Fix up some more rubocop offenses

## Details

Some auto-fixed, some manually fixed. Approx 70 failures have been fixed and the config has been regenerated

## Motivation and Context

Make the codebase better, one offense at a time.

## How Has This Been Tested?

CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactoring (cleanup of codebase without changing any existing functionality)
- [ ] Update documentation

## Checklist:

- [ ] I've added tests for my code
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: lib/aruba/platforms/command_monitor.rb [hunk 1]
```diff
@@ -22,6 +22,10 @@ def nil?
       def method_missing(*)
         fail NoCommandHasBeenStoppedError, 'No last command stopped available'
       end
+
+      def respond_to_missing?(*)
```
Comment (mvz) [inline lib/aruba/platforms/command_monitor.rb]: I suppose this is correct for now, but I'd really like to simplify this design.

File: lib/aruba/platform.rb [hunk 2]
```diff
@@ -4,18 +4,10 @@
 # Aruba
 module Aruba
   PLATFORM_MUTEX = Mutex.new
-end
+  PLATFORM = [Platforms::WindowsPlatform, Platforms::UnixPlatform].find(&:match?)
```
Comment (mvz) [inline lib/aruba/platform.rb]: Which cop makes you change `Platform` to `PLATFORM`? I feel this file should at least define `Aruba::Platform`. Renaming may even be a breaking change.

File: lib/aruba/api/core.rb [hunk 3]
```diff
@@ -15,7 +15,7 @@ module Core
 
       # Aruba Runtime
       def aruba
-        @_aruba_runtime ||= Runtime.new
+        @aruba ||= Runtime.new
```
[the remark this answers is not in the document]
Comment (luke-hill) [reply lib/aruba/api/core.rb]: I checked everywhere that module was included and there was no mention of `@aruba` anywhere in the entire codebase where that module was included

File: lib/aruba/platform.rb (hunk 2 above)
Comment (luke-hill) [reply lib/aruba/platform.rb]: Constants should be `SCREAMING_SNAKE_CASE`

Comment (mvz) [reply lib/aruba/platform.rb]: I think we should disable the cop for this particular case, since we're basically setting up a class here.

**Comment by mvz** [comment]
@luke-hill could you please rebase this on current master?

File: lib/aruba/platform.rb (hunk 2 above)
Comment (luke-hill) [reply lib/aruba/platform.rb]: It seems a proper convoluted way of doing things, but I'll alter this change for now and rebase.

**Comment by mvz** [comment]
@luke-hill I think you merged instead of rebasing? Let's wait for Travis right now and then fix that up.

**Comment by luke-hill** [comment]
I fixed this up and rebased with the latest master, but yeh now we have like 100 failures regarding environment.

I did a merge commit as it's easier than rebasing when your PR is >5 commits. But yeh lord knows what's gone on here. Have we removed something in `master`

File: lib/aruba/api/core.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (mvz) [reply lib/aruba/api/core.rb]: I think this change may be responsible for a lot of the build failures.

**Comment by mvz** [comment]
I don't have time now, but I can have a deeper look into this later in the week.

**Comment by luke-hill** [comment]
I've fixed up the issues and added a warning about that area. In theory that "shouldn't" have broken it, as we shouldn't be accessing a memo's iVar.

Still it's now all green and slightly better. It's also RtM if you're happy with it.

**Comment by mvz** [comment]
Thanks, @luke-hill!

**Pull request merged**